corehq/apps/reports/generic.py

```diff
@@ -609,9 +611,13 @@ def export_response(self):
         Intention: Not to be overridden in general.
         Returns the tabular export of the data, if available.
         """
-        temp = StringIO()
-        export_from_tables(self.export_table, temp, self.export_format)
-        return export_response(temp, self.export_format, self.export_name)
+        if self.exportable_all:
```

Comment:
I think this should check ```self.exportable``` since it we want to export the entire report.  I think ```self.exportable_all``` is actually a misleading name - when used, it sends the report via email instead of in a downloadable file.